5_data_munge/src/munge_data_files.R

```diff
@@ -0,0 +1,158 @@
+              n_day = n())
+  
+  dat_daily_meta <- filter(dat_reduced, !is.na(`ResultMeasure/MeasureUnitCode`)) %>%
+    select(-ResultMeasureValue, -`ActivityStartTime/Time`) %>%
+    group_by(MonitoringLocationIdentifier, ActivityStartDate, sample_depth) %>%
+    summarize_all(first)
+  
+  # bring it together
+  dat_daily <- left_join(dat_daily, dat_daily_meta)
+  
+  # save
+  data_file <- scipiper::as_data_file(out_ind)
+  saveRDS(dat_daily, data_file)
+  gd_put(out_ind, data_file)
+    
+}
+
+munge_wqp_withoutdepths <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+  dat <- readRDS(sc_retrieve(in_ind)) %>%
+    select(MonitoringLocationIdentifier, ActivityStartDate, ResultMeasureValue, 
+           `ResultMeasure/MeasureUnitCode`, `ActivityStartTime/Time`, ActivityMediaName, 
+           `ActivityDepthHeightMeasure/MeasureValue`, `ResultDepthHeightMeasure/MeasureValue`, 
+           `ActivityTopDepthHeightMeasure/MeasureValue`, `ActivityBottomDepthHeightMeasure/MeasureValue`) %>%
+    filter(!is.na(`ResultMeasure/MeasureUnitCode`)) %>%
+    filter(ActivityMediaName %in% 'Water') %>%
+    filter(is.na(`ActivityDepthHeightMeasure/MeasureValue`)&
+             is.na(`ResultDepthHeightMeasure/MeasureValue`)&
+             is.na(`ActivityTopDepthHeightMeasure/MeasureValue`)&
+             is.na(`ActivityBottomDepthHeightMeasure/MeasureValue`)) %>%
+    filter(ActivityStartDate > as.Date('1800-01-01') & ActivityStartDate < as.Date('2020-01-01'))
+  
+ 
+  # now reduce to daily measures
+  # first clean up obviously bad data
+  dat_reduced <- filter(dat, ResultMeasureValue > min_value) %>%
+    filter(grepl('deg C|deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE)) %>%
+    mutate(ResultMeasureValue = ifelse(grepl('deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE),
+                                       f_to_c(ResultMeasureValue), ResultMeasureValue)) %>%
+    mutate(`ResultMeasure/MeasureUnitCode` = 'deg C') %>%
+    filter(ResultMeasureValue < max_value)
+  
+  dat_daily <- group_by(dat_reduced, MonitoringLocationIdentifier, ActivityStartDate) %>%
+    summarize(temperature_mean_daily = mean(ResultMeasureValue),
+              activity_start_times = paste(`ActivityStartTime/Time`, collapse = ', '),
+              n_day = n())
+  
+  # save
+  data_file <- scipiper::as_data_file(out_ind)
+  saveRDS(dat_daily, data_file)
+  gd_put(out_ind, data_file)
+  
+}
+
+munge_nwis_uv <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+  nwis_df <- readRDS(sc_retrieve(in_ind)) %>%
+    filter(!is.na(temp_value), !is.na(dateTime)) %>%
+    filter(grepl('A', cd_value, ignore.case = FALSE)) %>%
+    filter(temp_value > min_value,
+           temp_value < max_value) %>%
+    mutate(dateTime = as.Date(dateTime)) %>%
+    group_by(site_no, col_name, dateTime) %>%
+    summarize(temperature_mean_daily = round(mean(temp_value), 1), n_day = n())
```

Comment:
Are all NWIS temperatures only reported to tenth-degree precision? It probably doesn't matter much for these daily values, but I saw unnecessary choppiness in sub-daily timeseries when I was doing metabolism modeling. If we could avoid adding any superfluous rounding here that might be nice...could round to the minimum number of sig figs among the values in `temp_value`, for example.

Reply:
Am going to round to 3 digits to give a bit of buffer.